jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -203,7 +203,24 @@ default void close()
      * filled or -1 if EOF is read or the input is shutdown.
      * @throws IOException if the endpoint is closed.
      */
-    int fill(ByteBuffer buffer) throws IOException;
+    default int fill(ByteBuffer buffer) throws IOException
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * <p>Receives data into the given buffer from the returned address.</p>
+     * <p>This method should be used to receive UDP data, in conjunction
+     * with {@link DatagramChannelEndPoint}.</p>
+     *
+     * @param buffer the buffer to fill with data
+     * @return the peer address that sent the data
+     * @throws IOException if the receive fails
+     */
+    default SocketAddress receive(ByteBuffer buffer) throws IOException
```

Comment:
So why can't the `receive` and `send` methods be moved to a sub-interface of `EndPoint` and in the few places they are needed then the user downcasts to a `DatagramEndPoint` ?

I know that is clunkier, but as we are likely to change the `EndPoint` abstraction in jetty-12 is seams wrong to extend it only to then change it.